module/input/SpeechIntent/src/SpeechIntent.cpp

```diff
@@ -0,0 +1,405 @@
+                              "en",
+                              "recognize-intent",
+                              NULL};
+
+        SpawnedProcess process = {};
+        if (spawn_process(process, args)) {
+            write(process.stdin, input, strlen(input));
+
+            bytes_read = read(process.stdout, buffer, sizeof(buffer) - 1);
+            if (bytes_read == -1) {
+                NUClear::log<NUClear::FATAL>(
+                    fmt::format("({}:{}) failed to read from spawned process "
+                                "(voice2json recognize-intent) stdout - errno = {}",
+                                __FILE__,
+                                __LINE__,
+                                errno));
+            }
+            buffer[bytes_read] = 0;
+
+            output = strdup(buffer);
+        }
+        if (process.stdout)
+            close(process.stdout);
+        if (process.stderr)
+            close(process.stderr);
+        if (process.stdin)
+            close(process.stdin);
+
+        return output;
+    }
+
+    // write out the correct format for voice2json to consume
+    // when voice2json is called using the "transcribe-wav --input-size" flags
+    // it excepts the first line to be the file size (as ascii)
+    // followed by the wav file after the newline
+    bool write_audio_to_file(int fd, char* filename) {
+        ssize_t write_res = 0;
+        if (access(filename, F_OK) != 0) {
+            return false;
+        }
+
+        FILE* file = fopen(filename, "rb");
+        fseek(file, 0, SEEK_END);
+        int file_size = ftell(file);
+        fseek(file, 0, SEEK_SET);
+        assert(file_size != -1);
+
+        char* file_buffer = (char*) malloc(file_size);
+        fread(file_buffer, file_size, 1, file);
+        fclose(file);
+
+        char file_size_buf[64];
+        sprintf(file_size_buf, "%d\n", file_size);
+
+        write_res = write(fd, file_size_buf, strlen(file_size_buf));
+        if (write_res == -1) {
+            NUClear::log<NUClear::FATAL>(fmt::format("({}:{}) write failed, errno = {}", __FILE__, __LINE__, errno));
+            free(file_buffer);
+            return false;
+        }
+
+        write_res = write(fd, file_buffer, file_size);
+        if (write_res == -1) {
+            NUClear::log<NUClear::FATAL>(fmt::format("({}:{}) write failed, errno = {}", __FILE__, __LINE__, errno));
+            free(file_buffer);
+            return false;
+        }
+        assert(write_res == file_size);
+
+        free(file_buffer);
+
+        return true;
+    }
+
+
+    void print_intent(const SpeechIntentMsg& msg) {
+        NUClear::log(fmt::format("text = {}", msg.text.data()));
+        NUClear::log(fmt::format("intent = {}", msg.intent.data()));
+        NUClear::log("slots");
+        for (Slot slot : msg.slots) {
+            NUClear::log(fmt::format("{} = {}", slot.name.data(), slot.value.data()));
+        }
+    }
+
+    void SpeechIntent::init() {
```

Comment:
Do we use init functions elsewhere in the code base? Not saying it's wrong, just seems against the grain